.github/workflows/ci.yml

```diff
@@ -0,0 +1,107 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: 🛑 Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: ⬇️ Checkout
+        uses: actions/checkout@v2
+
+      - name: ⎔ Setup node for problem matcher
+        if: github.event.pull_request
+        uses: actions/setup-node@v2
+
+      - name: 📥 Download deps
+        run: npm install
+
+      - name: 🧪 Run lint
+        run: npm run lint
+
+  build:
+    name: Build
+    runs-on: ${{ matrix.os }}
+    needs: lint
+    strategy:
+      fail-fast: false
+      matrix:
+        node:
+          - 12
+          - 14
+          - 16
+          - 17
+        os:
+          - ubuntu-latest
+          - macos-latest
+          - windows-latest
+        eslint:
+          - 7
+          - 8
+
+    steps:
+      - name: 🛑 Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: ⬇️ Checkout
+        uses: actions/checkout@v2
+
+      - name: ⎔ Setup node ${{ matrix.node }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node }}
+
+      - name: 📥 Download deps
+        run: npm install
+
+      - name: Install ESLint 7
+        if: matrix.eslint == 7
+        run: npm install --no-save eslint@7 eslint-config-canonical@24.4.4
```

Comment:
Don't really understand the need for the `eslint-config-conanical` here, but it was used in Travis, and 2 tests failed without it here https://github.com/nschonni/eslint-plugin-jsdoc/runs/4591072042?check_suite_focus=true#step:8:63

Reply:
The `check-examples` rule (which has had to be disabled for ESLint 8 tests, but should work for ESLint 7 still) has a behavior of checking the actual current package's eslintrc (i.e., that of `eslint-plugin-jsdoc`).

I don't recall, however, whether it fails because of more recent expectations breaking the tests (I don't think so) or because of some ESlint interaction problem. Sorry not of more help.

Reply:
Could you take a look at the current `.travis.yml` file--which resolves some issues we were having--and see about applying the conditional logic there?

`eslint-config-canonical` can now be pegged to `32.1.0`, but we also need to peg `eslint-plugin-mocha` to version `9` because it [no longer supports Node 12](https://github.com/lo1tuma/eslint-plugin-mocha/releases/tag/10.0.0)

I also added the `--engine-strict` check on the latest Node 16 version to ensure at least our development tools could all run on latest Node, but I couldn't add this to the earlier versions because as mentioned the latest linting tools won't work on Node 12. That also means we don't want linting to run in every environment--Node 16 should be enough (and also `npm run build`) is only really needed in Node 16 I think, as the critical item to check is that the tests work in all environments.